pkg/services/apiserver/options/extra.go

```diff
@@ -48,8 +48,15 @@ func (o *ExtraOptions) ApplyTo(c *genericapiserver.RecommendedConfig) error {
 	}); err != nil {
 		return err
 	}
-	// TODO: klog isn't working as expected, investigate - it logs some of the time
+	// if verbosity is 8+, response bodies will be logged. versboity of 7 should then be the max
+	if o.Verbosity > 7 {
+		o.Verbosity = 7
+	}
 	klog.SetSlogLogger(logger)
+	// at this point, the slog will be the background logger. set it as the default logger, as setting solely slog above
+	// won't update the verbosity because it is set as a contextual logger, and that function says "such a logger cannot
+	// rely on verbosity checking in klog"
+	klog.SetLogger(klog.Background())
```

Comment:
wonder if that's what was the cause of missed logs that I marked here in this comment: https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/options/extra.go#L51

maybe that can be removed if that is resolved now.

Reply:
ha, you already removed it, good!